src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -550,74 +589,159 @@ private void sendCitizenPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasN
+    private void sendCitizenPackets(@NotNull Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
     {
-        if (permissions.isDirty() || hasNewSubscribers)
+        if (isCitizensDirty || hasNewSubscribers)
         {
-            subscribers
-                    .stream()
-                    .filter(player -> permissions.isDirty() || !oldSubscribers.contains(player)).forEach(player -> {
-                Permissions.Rank rank = getPermissions().getRank(player);
-                MineColonies.getNetwork().sendTo(new PermissionsMessage.View(this, rank), player);
-            });
+            for (@NotNull CitizenData citizen : citizens.values())
+            {
+                if (citizen.isDirty() || hasNewSubscribers)
+                {
+                    subscribers.stream()
+                      .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
+                      .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(this, citizen), player));
+                }
+            }
         }
     }
 
-    private void sendColonyViewPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    /**
+     * Sends packages to update the buildings.
+     *
+     * @param oldSubscribers    the existing subscribers.
+     * @param hasNewSubscribers the new subscribers.
+     */
+    private void sendBuildingPackets(@NotNull Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
     {
-        if (isDirty || hasNewSubscribers)
+        if (isBuildingsDirty || hasNewSubscribers)
         {
-            for (EntityPlayerMP player : subscribers)
+            for (@NotNull AbstractBuilding building : buildings.values())
             {
-                boolean isNewSubscriber = !oldSubscribers.contains(player);
-                if (isDirty || isNewSubscriber)
+                if (building.isDirty() || hasNewSubscribers)
                 {
-                    MineColonies.getNetwork().sendTo(new ColonyViewMessage(this, isNewSubscriber), player);
+                    subscribers.stream()
+                      .filter(player -> building.isDirty() || !oldSubscribers.contains(player))
+                      .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewBuildingViewMessage(building), player));
                 }
             }
         }
     }
 
-    private static boolean hasNewSubscribers(Set<EntityPlayerMP> oldSubscribers, Set<EntityPlayerMP> subscribers)
+    /**
+     * Sends packages to update the fields.
+     *
+     * @param oldSubscribers    the existing subscribers.
+     * @param hasNewSubscribers the new subscribers.
+     */
+    private void sendFieldPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
     {
-        for (EntityPlayerMP player : subscribers)
+        if (isFieldsDirty && !isBuildingsDirty || hasNewSubscribers)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)